core/node/libp2p/rcmgr.go

```diff
@@ -324,45 +308,68 @@ func NetSetLimit(mgr network.ResourceManager, scope string, limit config.Resourc
 		return nil
 	}
 
+	cfg, err := repo.Config()
+	if err != nil {
+		return fmt.Errorf("reading config to set limit: %w", err)
+	}
+
+	setConfigLimit := func(f func(c *rcmgr.BasicLimiterConfig)) {
+		if cfg.Swarm.ResourceMgr.Limits == nil {
+			cfg.Swarm.ResourceMgr.Limits = &rcmgr.BasicLimiterConfig{}
+		}
+		f(cfg.Swarm.ResourceMgr.Limits)
+	}
+
 	switch {
 	case scope == config.ResourceMgrSystemScope:
-		err := mgr.ViewSystem(func(s network.ResourceScope) error {
+		err = mgr.ViewSystem(func(s network.ResourceScope) error {
 			return setLimit(s)
 		})
-		return err
+		setConfigLimit(func(c *rcmgr.BasicLimiterConfig) { c.System = &limit })
 
 	case scope == config.ResourceMgrTransientScope:
-		err := mgr.ViewTransient(func(s network.ResourceScope) error {
+		err = mgr.ViewTransient(func(s network.ResourceScope) error {
 			return setLimit(s)
 		})
-		return err
+		setConfigLimit(func(c *rcmgr.BasicLimiterConfig) { c.Transient = &limit })
 
 	case strings.HasPrefix(scope, config.ResourceMgrServiceScopePrefix):
-		svc := scope[4:]
-		err := mgr.ViewService(svc, func(s network.ServiceScope) error {
+		svc := strings.TrimPrefix(scope, config.ResourceMgrServiceScopePrefix)
+		err = mgr.ViewService(svc, func(s network.ServiceScope) error {
 			return setLimit(s)
 		})
-		return err
+		setConfigLimit(func(c *rcmgr.BasicLimiterConfig) { c.Service[svc] = limit })
 
 	case strings.HasPrefix(scope, config.ResourceMgrProtocolScopePrefix):
-		proto := scope[6:]
-		err := mgr.ViewProtocol(protocol.ID(proto), func(s network.ProtocolScope) error {
+		proto := strings.TrimPrefix(scope, config.ResourceMgrProtocolScopePrefix)
+		err = mgr.ViewProtocol(protocol.ID(proto), func(s network.ProtocolScope) error {
 			return setLimit(s)
 		})
-		return err
+		setConfigLimit(func(c *rcmgr.BasicLimiterConfig) { c.Service[proto] = limit })
 
 	case strings.HasPrefix(scope, config.ResourceMgrPeerScopePrefix):
-		p := scope[5:]
-		pid, err := peer.Decode(p)
+		p := strings.TrimPrefix(scope, config.ResourceMgrPeerScopePrefix)
+		var pid peer.ID
+		pid, err = peer.Decode(p)
 		if err != nil {
 			return fmt.Errorf("invalid peer ID: %q: %w", p, err)
 		}
 		err = mgr.ViewPeer(pid, func(s network.PeerScope) error {
 			return setLimit(s)
 		})
-		return err
+		setConfigLimit(func(c *rcmgr.BasicLimiterConfig) { c.Service[p] = limit })
```

Comment:
Same, needs `peer:<id>` needs sharness tests.


```suggestion
		setConfigLimit(func(c *rcmgr.BasicLimiterConfig) { c.Peer[p] = limit })
```